crates/model/src/order.rs

```diff
@@ -359,33 +360,91 @@ impl OrderData {
 
 /// An order as provided to the POST order endpoint.
 #[serde_as]
-#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
+#[schema(example = json!({
+  "sellToken": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "buyToken": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "receiver": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "sellAmount": "1234567890",
+  "buyAmount": "1234567890",
+  "validTo": 0,
+  "feeAmount": "1234567890",
+  "kind": "buy",
+  "partiallyFillable": true,
+  "sellTokenBalance": "erc20",
+  "buyTokenBalance": "erc20",
+  "signingScheme": "eip712",
+  "signature": "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+  "from": "0x6810e776880c02933d47db1b9fc05908e5386b96",
+  "quoteId": 0,
+  "appData": "{\"version\":\"0.9.0\",\"metadata\":{}}",
+  "appDataHash": "0x0000000000000000000000000000000000000000000000000000000000000000"
+}))]
 pub struct OrderCreation {
     // These fields are the same as in `OrderData`.
+    /// Address of token sold.
+    #[schema(value_type = String)]
     pub sell_token: H160,
+    /// Address of token bought.
+    #[schema(value_type = String)]
     pub buy_token: H160,
+    /// An optional address to receive the proceeds of the trade instead of the
+    /// `owner` (i.e. the order signer).
     #[serde(default)]
+    #[schema(value_type = Option<String>)]
     pub receiver: Option<H160>,
+    /// Amount of `sellToken` to be sold in atoms.
     #[serde_as(as = "HexOrDecimalU256")]
+    #[schema(value_type = String)]
     pub sell_amount: U256,
```

Comment:
For things like `U256` and `H160` I think it makes sense to add a new type that implements `ToSchema` and the sometimes custom deserialization logic. This will be annoying for the types in the orderbook because we used the API types in the business logic but eventually we should have a separation of concerns and the introduction of utoipa seems like a good option.